lib/resty/http_connect.lua

```diff
@@ -160,22 +174,113 @@ local function connect(self, options)
         proxy_port = proxy_uri_t[3]
     end
 
+    local cert_hash
+    -- fallback to non-mTLS when any error
+    repeat
+        if not ssl or not ssl_client_cert or not ssl_client_priv_key then
+            break
+        end
+
+        local cert_type = type(ssl_client_cert)
+        local key_type = type(ssl_client_priv_key)
+
+        if cert_type ~= "cdata" then
+            ngx_log(ngx_WARN, "bad ssl_client_cert: cdata expected, got ", cert_type)
+            break
+        end
+
+        if key_type ~= "cdata" then
+            ngx_log(ngx_WARN, "bad ssl_client_priv_key: cdata expected, got ", key_type)
+            break
+        end
+
+        local status, res = xpcall(require_openssl_libs, debug.traceback)
```

Comment:
can't we load the modules once at the module level, and set a flag to check for availability of them? I find this a weird construct with an unnecessary intermediate table being returned.

Reply:
make sense.